api/user/schema.graphql

```diff
@@ -13,11 +13,46 @@ extend type Query {
 }
 
 extend type Mutation {
-  logout: Boolean
-  login(email: EmailAddress!, password: String!): User
-  signup(email: EmailAddress!, password: String!): User
-  forgotPassword(email: EmailAddress!): Boolean
-  changePassword(token: String!, id: ID!, newPassword: String!): User
+  logout: LogoutPayload
+  login(email: EmailAddress!, password: String!): LoginPayload
+  signup(email: EmailAddress!, password: String!): SignupPayload
+  forgotPassword(email: EmailAddress!): ForgotPasswordPayload
+  changePassword(
+    token: String!
+    id: ID!
+    newPassword: String!
+  ): ChangePasswordPayload
+}
+
+union LoginPayload = UserReturned | InputValidationProblem
+
+union ChangePasswordPayload = UserReturned | ExpiredTokenProblem
+
+union SignupPayload = UserReturned | InputValidationProblem
+
+type ForgotPasswordPayload {
+  result: Boolean!
+}
+
+type LogoutPayload {
+  result: Boolean!
+}
+
+type UserReturned {
+  user: User!
+}
+
+type InputValidationProblem {
+  problems: [InputFieldValidationProblem]!
+}
+
+type ExpiredTokenProblem {
+  problems: [InputFieldValidationProblem]!
+}
```

Comment:
```suggestion
type ExpiredTokenProblem {
  message: String!
}
```
since the name of the problem already conveys that it is about a certain problem with the token.